Plugins/NativeEngine/Source/IndexBuffer.cpp

```diff
@@ -3,7 +3,7 @@
 namespace Babylon
 {
     IndexBuffer::IndexBuffer(gsl::span<uint8_t> bytes, uint16_t flags, bool dynamic)
-        : m_bytes{bytes.data(), bytes.data() + bytes.size()}
+        : m_bytes{std::vector<uint8_t>{bytes.data(), bytes.data() + bytes.size()}}
```

Comment:
nit:
```suggestion
        : m_bytes{{bytes.data(), bytes.data() + bytes.size()}}
```

Reply:
I haven't tried to actually run this, but this looks like it's going to construct the `std::vector` outside the `std::optional`, then move (or copy, not sure how it'll handle the R-value-ness) that vector into the optional. Do we think it's better to do this, or is it better to construct the `std::optional` empty and `emplace` the vector inside in the body of the constructor? If this line moves instead of copies, it probably doesn't make much difference cost-wise, but thought it was worth asking.

Reply:
It has to move unless the implementation of `std::optional` is broken. The `std::vector` is unnamed which should end up as a rvalue reference.